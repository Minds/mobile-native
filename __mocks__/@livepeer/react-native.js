// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
