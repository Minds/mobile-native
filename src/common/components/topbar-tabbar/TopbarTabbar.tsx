import React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleProp,
  TextStyle,
  StyleSheet,
  ViewStyle,
} from 'react-native';
import ThemedStyles from '../../../styles/ThemedStyles';

export type TabType<T> = {
  id: T;
  title: string;
  subtitle?: string;
};

type PropsType<T> = {
  tabs: Array<TabType<T>>;
  current: T;
  onChange: (id: T) => void;
  titleStyle?: StyleProp<TextStyle>;
  subtitleStyle?: StyleProp<TextStyle>;
  tabStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
};

/**
 * Tab bar
 */
function TopbarTabbar<T>(props: PropsType<T>) {
  const theme = ThemedStyles.style;
  const tabStyle = [
    theme.paddingVertical2x,
    theme.marginHorizontal2x,
    theme.borderBottom4x,
  ];

  return (
    <View
      style={[
        theme.rowJustifyStart,
        theme.borderBottom,
        theme.borderPrimary,
        theme.paddingHorizontal2x,
        props.containerStyle,
      ]}>
      {props.tabs.map((tab, i) => (
        <TouchableOpacity
          onPress={() => props.onChange(tab.id)}
          key={i}
          style={[
            tabStyle,
            props.tabStyle,
            tab.id === props.current
              ? theme.borderTab
              : theme.borderTransparent,
          ]}>
          <Text
            style={[
              theme.fontL,
              tab.id === props.current
                ? theme.colorPrimaryText
                : theme.colorSecondaryText,
              props.titleStyle,
            ]}>
            {tab.title}
          </Text>
          {!!tab.subtitle && (
            <Text
              style={[
                theme.fontL,
                theme.colorSecondaryText,
                styles.subtitle,
                props.subtitleStyle,
              ]}>
              {tab.subtitle}
            </Text>
          )}
        </TouchableOpacity>
      ))}
    </View>
  );
}

export default TopbarTabbar;

const styles = StyleSheet.create({
  subtitle: {
    paddingVertical: 2,
  },
});
