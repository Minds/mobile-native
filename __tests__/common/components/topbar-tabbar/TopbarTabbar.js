import 'react-native';
import React from 'react';
import TopbarTabbar from '../../../../src/common/components/topbar-tabbar/TopbarTabbar';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tabs = [
    { id: 'tokens', title: 'Tokens', subtitle: '711' },
    { id: 'usd', title: 'USD', subtitle: '$123' },
    { id: 'eth', title: 'Ether', subtitle: '133' },
    { id: 'btc', title: 'Bitcoin', subtitle: '153' },
  ];
  const topbarTabbar = renderer.create(<TopbarTabbar tabs={tabs} />).toJSON();
  expect(topbarTabbar).toMatchSnapshot();
});
