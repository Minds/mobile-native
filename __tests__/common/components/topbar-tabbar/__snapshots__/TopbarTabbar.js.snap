// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    onPress={[Function]}
    style={
      Array [
        Array [
          undefined,
          undefined,
          undefined,
        ],
        undefined,
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Tokens
    </Text>
    <Text
      style={
        Array [
          undefined,
          undefined,
          Object {
            "paddingVertical": 2,
          },
          undefined,
        ]
      }
    >
      711
    </Text>
  </View>
  <View
    onPress={[Function]}
    style={
      Array [
        Array [
          undefined,
          undefined,
          undefined,
        ],
        undefined,
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      USD
    </Text>
    <Text
      style={
        Array [
          undefined,
          undefined,
          Object {
            "paddingVertical": 2,
          },
          undefined,
        ]
      }
    >
      $123
    </Text>
  </View>
  <View
    onPress={[Function]}
    style={
      Array [
        Array [
          undefined,
          undefined,
          undefined,
        ],
        undefined,
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Ether
    </Text>
    <Text
      style={
        Array [
          undefined,
          undefined,
          Object {
            "paddingVertical": 2,
          },
          undefined,
        ]
      }
    >
      133
    </Text>
  </View>
  <View
    onPress={[Function]}
    style={
      Array [
        Array [
          undefined,
          undefined,
          undefined,
        ],
        undefined,
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Bitcoin
    </Text>
    <Text
      style={
        Array [
          undefined,
          undefined,
          Object {
            "paddingVertical": 2,
          },
          undefined,
        ]
      }
    >
      153
    </Text>
  </View>
</View>
`;
